src/annepro2.rs

```diff
@@ -50,16 +50,17 @@ pub enum AP2FlashError {
     OtherError,
 }
 
-pub fn flash_firmware<R: std::io::Read>(target: AP2Target, base: u32, file: &mut R, interface: i32, boot: bool) -> std::result::Result<(), AP2FlashError> {
+pub fn flash_firmware<R: std::io::Read>(target: AP2Target, base: u32, file: &mut R, boot: bool) -> std::result::Result<(), AP2FlashError> {
     match HidApi::new() {
         Ok(api) => {
             for dev in api.device_list() {
                 println!("HID Dev: {:04x}:{:04x} if: {} {:?}", dev.vendor_id(), dev.product_id(), dev.interface_number(), dev.product_string());
             }
             let dev = api.device_list().find(|dev| {
                 dev.vendor_id() == ANNEPRO2_VID &&
-                    ((dev.product_id() == PID_C15 && dev.interface_number() == 1) || (dev.product_id() == PID_C18 && dev.interface_number() == interface))
-            }).expect("No device found. (If you have the c18 revision, you must specify the correct interface number using -i=<number>)");
+                    ((dev.product_id() == PID_C15 && dev.interface_number() == 1) ||
```

Comment:
First, sorry for putting this off for so long. 

I think I mentioned this in the discord chat, but I will reiterate it here. I believe the proper way to fix this is by reading `dev.usage_page()` and `dev.usage()`. With my new knowledge about USB, this is the correct way to identify a HID device by it's function.